### PR TITLE
OS: Keep only c++11 implementations and remove YARP_EXPERIMENTAL_CXX11 option

### LINKED ISSUE #1035: yarp segfault on exit when YARP_EXPERIMENTAL_CXX11=true
Observed on Win32, compiled with VS2013 ULTIMATE.
Almost all yarp applications (including the yarp name server) are crashing after a CTRL+C.

Still untested on Linux.

### PR DESCRIPTION
**Pull request by drdanz**
Fixes #1035 and #1061

### CODE REVIEW

**Comment by drdanz** [comment]
Currently broken on windows in Debug configuration

File: src/libYARP_OS/src/ThreadImpl.cpp [hunk 1]
```diff
@@ -363,9 +327,7 @@ bool ThreadImpl::isClosing()
 
 bool ThreadImpl::isRunning()
 {
-    threadMutex->wait();
     bool b = active;
-    threadMutex->post();
     return b;
```
Comment (barbalberto) [inline src/libYARP_OS/src/ThreadImpl.cpp]: Can it be simply `return active`? (without defining `bool b`)

Comment (drdanz) [reply src/libYARP_OS/src/ThreadImpl.cpp]: :+1:

**Comment by drdanz** [comment]
> Can this update help us to get rid of ThreadImpl::init() ?

@barbalberto What's wrong with `ThreadImpl::init()`? You mean getting rid of the mutexes?

**Comment by barbalberto** [comment]
Yes, so we could use mutexes and threads without requiring `Network` to be initialized.

**Comment by drdanz** [comment]
@barbalberto wrote
> Yes, so we could use mutexes and threads without requiring Network to be initialized.

:+1: I think you can already use mutex and semaphores, but it is completely untested. For threads, we definitely need to get rid of `ThreadImpl::init()`

**Pull request merged**